### ISSUE TITLE
Broken CI

### ISSUE THREAD

**Issue by Yajo** [OCA/maintainer-quality-tools#626]
This repo is broken.

The tests exhibit some kind of race condition that sometimes makes builds fail.

You can see the failures in https://travis-ci.org/OCA/maintainer-quality-tools/builds/637822768, which are unrelated to the PR.

I just rebuilt one job in the master branch and it failed with the same failure: https://travis-ci.org/OCA/maintainer-quality-tools/jobs/623695176

The failure seems to come from around the combination of these code chunks, and has been probably introduced by @hugo-ad in https://github.com/OCA/maintainer-quality-tools/pull/434:

https://github.com/OCA/maintainer-quality-tools/blob/f5651848a8670ddddfe71a29e6639de315b1b826/travis/self_tests#L255-L257

https://github.com/OCA/maintainer-quality-tools/blob/ae6dfd29dfdf00a34a38bc4c75fea0a32e1bec94/travis/test_server.py#L409-L454

I have no idea on how to fix it, this is all the info I could get :confused:

**Comment by pedrobaeza** [comment]
@moylop260 you are one of the masters of this repo. Can you help?

**Comment by moylop260** [comment]
I @Yajo 
Thank for your issue

FYI I see the error here:
 - https://travis-ci.org/OCA/maintainer-quality-tools/jobs/623695176#L756

I have created the following PR in order to verify it:
 - https://github.com/OCA/maintainer-quality-tools/pull/627

**Comment by Yajo** [comment]
Just in case it helps, let me explain how I got to the conclusion I said in https://github.com/OCA/maintainer-quality-tools/issues/626#issue-552863044.

AFAICS, comparing the :x: job with the latest :heavy_check_mark: one, the main difference is here:

[Job that failed](https://travis-ci.org/OCA/maintainer-quality-tools/jobs/623695176#L722-L726):

```
test_modules_to_preinstall (__main__.MainTest) ... ok
test_pylint_check (__main__.MainTest)
Testing empty paths and pylint_run fix of: ... skipped 'Set LINT_CHECK'
test_travis_helpers (__main__.MainTest) ... ----------
ok
```

[Job that succeeded](https://travis-ci.org/OCA/maintainer-quality-tools/jobs/614129658):

```
test_modules_to_preinstall (__main__.MainTest) ... ok
----------
test_pylint_check (__main__.MainTest)
Testing empty paths and pylint_run fix of: ... skipped 'Set LINT_CHECK'
test_travis_helpers (__main__.MainTest) ... ('\x1b[1;31mFAIL\x1b[0;m', 'Command exited with code -9')
('\x1b[1;31mFAIL\x1b[0;m', 'Found 1 lines with errors')
Module test summary
('\x1b[1;31mFAIL\x1b[0;m', 'broken_module')
Expected 2 errors, found 1!
ok
```

I have searched and there seems to be nothing that exits with -9, but the closest one seems to be: https://github.com/OCA/maintainer-quality-tools/blob/f5651848a8670ddddfe71a29e6639de315b1b826/travis/self_tests#L255-L257

It's a weird way to kill the server. It would have made more sense to start it and stop it in the same test, or in the setup/teardown methods. Also, it seems like the server is expecting 1 more error than what it finds. That's why I said it seems a race condition (after all, nothing has changed recently in those parts of the code).

I hope it helps.
